src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1084,28 +1069,18 @@ ExpressionPtr Visitor::visitTypesafe(Parser::PrimaryExpressionContext* ctx) {
     return visitTypesafe(ctx->brackettedExpression());
   }
 
-  // TODO<joka921> Refactor s.t. try/catch becomes if/else here
   if (ctx->numericLiteral()) {
-    auto literalAny = visitNumericLiteral(ctx->numericLiteral());
-    try {
-      auto intLiteral = literalAny.as<unsigned long long>();
-      return std::make_unique<sparqlExpression::IntExpression>(
-          static_cast<int64_t>(intLiteral));
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<long long>();
-      return std::make_unique<sparqlExpression::IntExpression>(
-          static_cast<int64_t>(intLiteral));
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<double>();
-      return std::make_unique<sparqlExpression::DoubleExpression>(
-          static_cast<double>(intLiteral));
-    } catch (...) {
-    }
-    AD_CHECK(false);
+    auto integralWrapper = [](int64_t x) {
```

Comment:
in this function you again can move all the cases that are just `visitTypesafe` to the end and make them `visitAlternative`.